### DISCUSSION TITLE
最近貌似有人在攻击 better-cloudflare-ip

### DISCUSSION THREAD

**Discussion by Ansen** [XIU2/CloudflareSpeedTest#340] [讨论]
最近有人在攻击 https://github.com/badafans/better-cloudflare-ip

![](https://i.imgur.com/D293kUT.png)

**Comment by XIU2** [comment]
我见到过类似讨论，确实一些人对 better-cloudflare-ip 项目新版本的原理存在争论。

不过这个你不应该去 better-cloudflare-ip 项目发 Issues 么，怎么跑到我这里了 ？

还是说你认为是我在攻击 better-cloudflare-ip ？。。。

****

我刚发布本项目没多长时间，就在 Github 的 Explore 推荐中意外发现了 better-cloudflare-ip，当时还去拜访交流过 [#19](https://github.com/badafans/better-cloudflare-ip/issues/19) ~

**Comment by Ansen** [comment]
> 
额，作者误会了，给你提issuse只是想让你知道这事，没其它意思，也没有认为是你在攻击，我还帮你解释来着

**Comment by XIU2** [comment]
一开始 better-cloudflare-ip 和 CloudflareST 原理一样，都是批量 Ping + 下载测速。
不过后来 better-cloudflare-ip 新版本改成了另一种方案，我们两个项目算是区分开了，不过当时我去学习原理的时候，就已经见到有人讨论隐私相关问题了。。。

总之这种新方案**有利有弊**，再加上设计起来比较折腾、麻烦，所以我也没有跟进（当然我 Golang 只是临时学的，技术不足也是主要原因之一 ...），依然选择维持现状，延续传统的简单粗暴方案（也算是多样化，让用户可以有更多选择？

**Comment by Ansen** [comment]
> 

隐私这个人觉得有点鸡蛋挑骨头，先不说 better-cloudflare-ip 已经[明确告知了会收集数据](https://github.com/badafans/better-cloudflare-ip/commit/f87c3c222654bb5c6ba31b9d15a390f123526312#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)，

但凡看一下脚本都能知道传了哪些数据，还可以自行注释掉上传命令

我个人是比较支持多样化的，实际上我两个项目都在用，能满足我不同的需求

另外作者太谦虚了，源阅读忠实粉丝路过

**Comment by XIU2** [comment]
对方攻击发送的内容，看似是在给我的项目引流，实则会让大家误以为是我干的呀。。。

人在家中坐，锅从天上来，这就搞得我很尴尬了，都不知道该不该去解释，但我感觉去解释的话反而显得很。。。

你发的这个 Issues 的时候，看到标题和图片内容，我**第一反应**都认为是我干的（常见的**受益方嫌疑论**，因此我不用想就知道其他人看到后的第一反应了），然后再一想又误以为该 Issues 想表达的意思是你认为是我干的（然后就是上面的交流了）。。。

**Comment by Ansen** [comment]
抱歉，是我没说清楚😂

我只是想让你知道这事

再说这也没啥引流的必要，国内开源都是吃力不讨好的事

**Comment by XIU2** [comment]
为了避免一直有人跑来问，我把这个 Issues 置顶了。

**Comment by badafans** [comment]
我关停better-cloudflare-ip项目与XIU2的这个项目无关，只是觉得个人做吃力不讨好的事情让人不舒服。

**Comment by Ansen** [comment]
> 我关停better-cloudflare-ip项目与XIU2的这个项目无关，只是觉得个人做吃力不讨好的事情让人不舒服。

国内做开源就是吃力不讨好居多

**Comment by XIU2** [comment]
@badafans 
我刚准备在你项目里发个 Issues 问问啥情况，正在纠结措辞如何说比较好的时候，没想到你专门跑过来解释了。。。
只能默默给你点个赞了。。。
